### ISSUE TITLE
Unable to go back from wallet view using back button [Android 13]

### ISSUE THREAD

**Issue by Uni-verse** [brave/brave-browser#27787]
## Description 
After selecting wallet from the hamburger menu, user is unable to go back to NTP. This is the case in all wallet views. User has to background/foreground or relaunch.

I have a Samsung Tablet with Android version 12 and it doesn't reproduce.

## Steps to reproduce 

   1. Install 1.46/1.47/1.48
   2. Launch brave
   3. Open menu, select wallet
   4. Try to navigate back using back button

## Actual result 
Does not go back to NTP

https://user-images.githubusercontent.com/17885425/212178827-6ffd6086-e4c7-48a4-95d2-3f3d2fda015e.mp4

## Expected result
Should go back to NTP

## Issue reproduces how often 
Easily

## Version/Channel Information:

- Can you reproduce this issue with the current Play Store version? Yes
- Can you reproduce this issue with the current Play Store Beta version? Yes
- Can you reproduce this issue with the current Play Store Nightly version? Yes

## Device details
   - Install type (ARM, x86): ARM
   - Device type (Phone, Tablet, Phablet): Samsung GS 21
   - Android version: 13
   
## Brave version
1.46.x, 1.47.x, 1.48.x, 1.49.x

### Website problems only
- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards?
- Is the issue reproducible on the latest version of Chrome?

**Comment by srirambv** [comment]
Good catch. Definitely Android 13 issue as I haven't seen it happen on my Android 12 devices

**Comment by kjozwiak** [comment]
@Uni-verse it's a bit different on the `Pixel 6` as  I don't have a bottom menu with a back button. I'm pretty sure with the `Pixel 6`, you basically need to swipe but that's broken as well. @srirambv should you be able to swipe back to the NTP/last page you were viewing before opening the wallet? I don't see any way back to the browser after opening the Brave Wallet.

* `Pixel 6` running `Android 13` using `1.47.172 Chromium: 109.0.5414.87` - **Reproduced** (can't swipe back/stuck in wallet)
* `Samsung S10+` running `Android 12` using `1.47.172 Chromium: 109.0.5414.87` - **Couldn't reproduce**

Example of the issue re: swiping not working:

https://user-images.githubusercontent.com/2602313/212460113-b6cffdd8-f7d0-4f4b-9560-4ed7c6858c76.mp4

**Comment by lukemulks** [comment]
+1 this, encountered on Samsung Fold3 on Android v13.

 1.47.175 Chromium: 109.0.5414.87 (Official Build) (64-bit) 

*Also tested the following journeys, all with the same result of being unable to go "back" to browsing from Wallet UX:* 

- toggle network to Solana Mainnet Beta; attempts to use OS back button do not navigate user back to browsing UX.

- toggle network to Ethereum Mainnet; attempts to use OS back button do not navigate user back to browsing UX.

- switch from Portfolio => Accounts view; attempts to use OS back button do not navigate user back to either the previous Portfolio UI, or browsing UX outside of the Wallet. 

-----

If the user has navigated from wallet/Portfolio => wallet/Accounts; the OS back button should navigate the user back to wallet/Portfolio as that screen and UI were where the user was in the journey, prior to being in Wallet/Accounts view. 

If the user is in Portfolio view and uses the OS back button; navigate the user to the browser tab they were active in prior to navigating to brave://wallet/portfolio

**Comment by srirambv** [comment]
Workaround for now is to background the app and relaunch which closes the wallet view.

**Comment by Pavneet-Sing** [comment]
Same community issue report: https://community.brave.com/t/android-how-to-switch-back-to-browser/464823

**Comment by kjozwiak** [comment]
The above requires `1.47.190` or higher for `1.47.x` verification 👍

**Comment by kjozwiak** [comment]
Moving this into `1.48.x` as we neve received another `C109` minor chromium bump that would have triggered another `1.47.x` release. We'll leave https://github.com/brave/brave-core/milestone/267 opened as there's code that landed in `1.47.x` but we'll close off the `1.47.x - Release #6` milestone and move all the issues within `1.48.x`.

The above can be verified with `1.48.156` for `1.48.x` as it's been merged into a `1.48.x` a while ago as per https://github.com/brave/brave-core/pull/16910.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Pixel 6` running `Android 13` using the following build(s):

```
Brave | 1.48.155 Chromium: 110.0.5481.77 (Official Build) (32-bit)
--- | ---
Revision | 65ed616c6e8ee3fe0ad64fe83796c020644d42af-refs/branch-heads/5481@{#839}
OS | Android 13; Build/TQ1A.230105.002
```

* ensured that you can go back to `Brave` by swiping while running through the `Brave Wallet` onboarding
* ensured that you can go back to `Brave` by swiping while within `Brave Wallet`
* ensured that you can go back to `Brave` by swiping while `Brave Wallet` has been locked
